cachito/workers/scm.py

```diff
@@ -124,9 +161,13 @@ def fetch_source(self):
         """Fetch the repo, create a compressed tar file, and put it in long-term storage."""
         # If it already exists and isn't corrupt, don't download it again
         archive_path = self.sources_dir.archive_path
-        if archive_path.exists() and tarfile.is_tarfile(archive_path):
+        if archive_path.exists():
             log.debug('The archive already exists at "%s"', archive_path)
-            return
+            try:
+                self._verify_archive()
+                return
+            except CachitoError:
+                log.debug('The archive at "%s" is invalid and will be re-created', archive_path)
```

Comment:
```suggestion
                log.error('The archive at "%s" is invalid and will be re-created', archive_path)
```
This could be warning too but I think it should not be `debug`